python/lsst/cell_coadds/typing_helpers.py

```diff
@@ -34,14 +34,14 @@ class ImageLike(Protocol):
     """
 
     @overload
-    def __getitem__(self: _S, slices: Tuple[slice, slice]) -> _S:
+    def __getitem__(self: _S, slices: tuple[slice, slice]) -> _S:
```

Comment:
I wonder if a return value of `Self` will work here now we have python 3.11 and you can drop the `_S`?